### PR TITLE
test case for config REST endpoint and app-defined data source

### PR DESCRIPTION
**Pull request by njr-11**
Add a test case that defines an app-defined data source and attempts to display its configuration via the /ibm/api/config REST endpoint.  Currently, much of this won't work because we haven't implemented it, so parts of the test are disabled.

### CODE REVIEW

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
#build

File: dev/com.ibm.ws.rest.handler.config_fat/test-applications/AppDefResourcesApp/src/test/resthandler/config/appdef/web/AppDefinedResourcesServlet.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                                     "createDatabase=create",
+                                     "onConnect=DECLARE GLOBAL TEMPORARY TABLE TEMP2 (COL1 VARCHAR(80)) ON COMMIT PRESERVE ROWS NOT LOGGED",
+                                     "queryTimeout=1m22s",
+                                     "reapTime=2200ms"
+                      })
+
+@SuppressWarnings("serial")
+@WebServlet(urlPatterns = "/AppDefinedResourcesServlet")
+public class AppDefinedResourcesServlet extends FATServlet {
+    //@Resource
```
Comment (alexsm82) [inline dev/com.ibm.ws.rest.handler.config_fat/test-applications/AppDefResourcesApp/src/test/resthandler/config/appdef/web/AppDefinedResourcesServlet.java]: Was this intentionally left commented out? Should there be a TODO?

Comment (njr-11) [reply dev/com.ibm.ws.rest.handler.config_fat/test-applications/AppDefResourcesApp/src/test/resthandler/config/appdef/web/AppDefinedResourcesServlet.java]: It was there to experiment with whether the processing of the app-defined data source would be deferred until the data source is looked up or injected, but was found to be unnecessary and can be removed.

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
Build failed due to infrastructure/connectivity issue. However, the commits for this pull were also included under #7483 and delivered as part of that pull, so I'm closing this.

**Pull request merged**